api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -6,7 +6,15 @@ class CartController < ::Spree::Api::V2::BaseController
           def create
             spree_authorize! :create, Spree::Order
 
-            order = spree_current_order || dependencies[:create_cart].call(user: spree_current_user, store: spree_current_store).value
+            order = if spree_current_order.present?
+                      spree_current_order
+                    else
+                      dependencies[:create_cart].call(
+                        user: spree_current_user,
+                        store: spree_current_store,
+                        currency: current_currency
+                      ).value
+                    end
```

Comment:
What do you think about rewriting it as follows?
```ruby
order   = spree_current_order if spree_current_order.present?
order ||= dependencies[...].call(....)
```

Reply:
I like the more verbose version more, but that's only mine personal preference ;)

Reply:
As for me, It's more better to use approach as @kushniryb said, looks more simple and readable